### PR TITLE
feat: add Lion and Automagic optimizers

### PR DESCRIPTION
**Pull request by saltysalrua**
## Summary

  - Add Lion optimizer support, including backend optimizer registry wiring, training schema fields, frontend labels, and
  CLI/schema tests.
  - Add Automagic optimizer support based on AI Toolkit-style per-parameter adaptive learning-rate masks, including
  `lr_scheduler=none` validation and dynamic LR monitor metrics.
  - Add `optimum-quanto` dependency for Automagic QBytesTensor compatibility.

### CODE REVIEW

**Comment by saltysalrua** [comment]
补充：还添加了cosine with warnup的调度器和Tlora
